app/assets/src/components/SampleUpload.jsx

```diff
@@ -129,7 +130,7 @@ class SampleUpload extends React.Component {
 
   toggleCheckBox(e) {
     this.setState({
-      omit_subsampling_checked: e.target.value == "true" ? false : true,
+      [e.target.id]: e.target.value == "true" ? false : true,
```

Comment:
what is the array [] state?  they should at least belong to a hash entry, not expose direct ids.